pkg/flow/adapter/xmltojsontransformation/adapter_test.go

```diff
@@ -56,35 +58,40 @@ func TestSink(t *testing.T) {
 			expectEvent: newCloudEvent(t, tJSONOutput1, cloudevents.ApplicationJSON),
 		},
 	}
+
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
-			svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-				body, err := ioutil.ReadAll(r.Body)
-				assert.NoError(t, err)
-				assert.Equal(t, tXML1, string(body))
-				fmt.Fprintf(w, "OK")
-			}))
-			defer svr.Close()
-
-			env := &envAccessor{
-				EnvConfig: adapter.EnvConfig{
-					Component: tCloudEventSource,
-					Sink:      svr.URL,
-				},
```

Comment:
I rewrote this test entirely — based on a similar test inside `xslttransformation/adapter_tests.go` — because it didn't test anything at all.

The adapter code wasn't even involved. We just spawned a httptest server (not even a CloudEvent receiver) which had nothing to do with the adapter, and which was asserting that the received body was equal to some constant (???)